xarray/core/computation.py

```diff
@@ -1141,6 +1141,91 @@ def earth_mover_distance(first_samples,
         return apply_array_ufunc(func, *args, dask=dask)
 
 
+def apply_to_dataset(func, obj, *args, **kwargs):
+    """apply a function expecting a Dataset to a xarray object
+
+    Parameters
+    ----------
+    func : callable
+        A function expecting a Dataset as its first parameter.
+    obj : DataArray or Dataset
+        The dataset to apply ``func`` to. If a ``DataArray``, convert it to a single
+        variable ``Dataset`` first.
+    *args, **kwargs
+        Additional arguments to ``func``
+
+    Returns
+    -------
+    DataArray or Dataset
+        The result of ``func(obj, *args, **kwargs)`` with the same type as ``obj``.
+
+    Notes
+    -----
+    If a ``DataArray``, the data variable of the temporary ``Dataset`` will have a
+    generic name. The original name will be restored for the result of the call.
```

Comment:
Feel free to ignore; I'm trying to make it obvious that most people don't need to worry about it.

```suggestion
    Internally, if a ``DataArray`` is passed, the data variable of the temporary
    ``Dataset`` will have a generic name. The original name will be restored
    for the result of the call.
```